routes/users.js

```diff
@@ -1,18 +1,74 @@
 const express = require("express");
 const router = express.Router();
 const jwt = require("jsonwebtoken");
+const User = require("../models/User");
+const authenticateToken = require("../middlewares/authToken");
+// const bcrypt = require("bcryptjs");
+const bcrypt = require("bcrypt");
+const saltRounds = 10;
 
-router.post(/login/, async (req, res) => {});
+router.post("/register", async (req, res) => {
+    const { username, password, role } = req.body;
 
-function authenticateToken(req, res, next) {
-    const authHeader = req.headers["authorization"];
-    const token = authHeader && authHeader.split(" ")[1];
-    if (token == null) return res.sendStatus(401);
+    if (!username || !password) {
+        return res.status(400).json({ message: "Missing username or password" });
+    }
 
-    jwt.verify(token, process.env.ACCESS_TOKEN_SECRET, (err, user) => {
-        console.log(err);
-        if (err) return res.sendStatus(403);
-        req.user = user;
-        next();
+    const user = await User.findOne({ username });
+    if (user) {
+        return res.status(400).json({ message: "User already exists" });
+    }
+
+    hash = bcrypt.hash(password, saltRounds, async function (err, hash) {
+        if (err) return res.status(500).json({ message: "Error hashing password" });
+
+        const newUser = new User({
+            username: username,
+            password: hash,
+            role: role,
+        });
+
+        try {
+            const savedUser = await newUser.save();
+            res.status(201).json(savedUser);
+        } catch (error) {
+            res.status(500).json({ message: error.message });
+        }
     });
-}
+});
+
+router.post("/login", async (req, res) => {
+    const { username, password } = req.body;
+
+    if (!username || !password) {
+        // ! wont get here since we have a frontend validation
+        return res.status(400).json({ message: "Missing username or password" });
+    }
+
+    const user = await User.findOne({ username });
+    if (!user) {
+        return res.status(400).json({ message: "User doesn't exist" });
+    }
+
+    bcrypt.compare(password, user.password, (err, isMatch) => {
+        if (err) return res.status(500).json({ message: "Error during password comparison" });
+        if (!isMatch) return res.status(400).json({ message: "Invalid credentials / Incorrect password" });
```

Comment:
Error message exposing details we don't want to expose. Try using `User or password incorrect. Try again`